### PR TITLE
add lyft cancel method

### PR DESCRIPTION
**Pull request by cpruijsen**

### CODE REVIEW

**Comment by cpruijsen** [comment]
`Error: Cannot find module '../../secret/config'`

File: src/server/models/Ride.js [hunk 1]
```diff
@@ -1,11 +1,8 @@
 import { Ride } from '../../db/Ride';
 import { User } from '../../db/User';
+import fetch from 'node-fetch';
 
-var fetch = require('node-fetch');
-var config = require('./../../../secret/config.js');
-var lyfthelper = require('./../utils/lyft-helper.js');
-var uberhelper = require('./../utils/uber-helper.js');
-
+const config = require('./../../../secret/config.js');
```
Comment (daredia) [inline src/server/models/Ride.js]: lets update this to check for env variables

File: src/server/utils/lyft-helper.js [hunk 2]
```diff
@@ -1,196 +1,285 @@
...
     })
-    .then(function (data) {
-      console.log('success posting user', data);
+    .then(data => {
+      // the responseMethod function returns an object with the parameters we need for subsequent operations only, and in a key-name generalised manner.
+      let response = phoneCodeAuth.responseMethod(data, userId);
+
+      // POST THE USER DATA TO OUR RELATIONAL DATABASE
+      // let dbpostURL = 'http://' + APIserver + '/users/updateOrCreate';
+      let dbpostURL = 'http://localhost:8080/users/updateOrCreate';
```
Comment (daredia) [inline src/server/utils/lyft-helper.js]: can we please get rid of this localhost reference and instead assign APIserver at the top of the file based on env? its actually already done like that on master right now, so this seems like a regression / omission of merging that code from master

File: src/server/utils/lyft-helper.js [hunk 3]
```diff
@@ -1,196 +1,285 @@
...
       return res.json();
     })
-    .then(function (data) {
-      console.log('success posting user', data);
+    .then(data => {
+      console.log('successful requestRide post LYFT', data);
+      let response = requestRide.responseMethod(data, tripDuration);
+
+      // let dbpostURL = 'http://' + APIserver + '/rides/' + rideId;
+      let dbpostURL = 'http://localhost:8080/rides/' + rideId;
```
Comment (daredia) [inline src/server/utils/lyft-helper.js]: another localhost reference to remove

File: src/server/utils/lyft-helper.js [hunk 4]
```diff
@@ -1,196 +1,285 @@
...
+    .then(data => {
+      console.log('successful cancelRide PUT LYFT', data);
+      let response = cancelRide.responseMethod(data);
 
-  })
-  .catch(function (err) {
-    console.log('error post of requestRide LYFT', err);
-  });
+      // let dbpostURL = 'http://' + APIserver + '/rides/' + rideId;
+      let dbpostURL = 'http://localhost:8080/rides/' + rideId;
```
Comment (daredia) [inline src/server/utils/lyft-helper.js]: another localhost reference to remove

File: src/server/utils/lyft-helper.js (hunk 2 above)
Comment (cpruijsen) [inline src/server/utils/lyft-helper.js]: sure - non used file btw

File: src/server/utils/lyft-helper.js (hunk 3 above)
Comment (cpruijsen) [inline src/server/utils/lyft-helper.js]: file is just for show

**Pull request merged**